apollo-ios/Sources/Apollo/MaxRetryInterceptor.swift

```diff
@@ -51,11 +99,47 @@ public class MaxRetryInterceptor: ApolloInterceptor {
     }
     
     self.hitCount += 1
-    chain.proceedAsync(
-      request: request,
-      response: response,
-      interceptor: self,
-      completion: completion
-    )
+    
+    // Apply exponential backoff delay if enabled and this is a retry (hitCount > 1)
+    if self.configuration.enableExponentialBackoff && self.hitCount > 1 {
+      let delay = calculateExponentialBackoffDelay()
+      DispatchQueue.main.asyncAfter(deadline: .now() + delay) { [weak self] in
```

Comment:
I don't remember the queueing strategy for the request chain but we should check that throwing this onto `main` is the right queue to be using.

Reply:
Thanks for pointing that out. I’m actually unsure which queue fits this pattern best, what do you think?” “I ended up using the main queue, but it’s harder to justify than I expected.

Reply:
That default `main` queue value on function parameters I believe is only actually used for the error or success callbacks, i.e.: a terminating queue. `kickoff`, `interceptAsync`, `proceedAsync`, or `retry` don't appear to be called on any specific queue. Those in turn are called from a network transport, and ultimately from client app code. So the originating queue is hidden from us.

I agree with @BobaFetters, if we can push this onto the current queue that would be best - I don't like the idea of reaching to Obj-C methods for that but it appears it's all we've got. If that doesn't work then `main` might be the next best choice.